core/src/transaction.rs

```diff
@@ -235,18 +239,19 @@ impl CellOutput {
                 .unwrap_or(0)
     }
 
-    pub fn destruct(self) -> (Capacity, Bytes, Script, Option<Script>) {
+    pub fn destruct(self) -> (Capacity, H256, Script, Option<Script>) {
         let CellOutput {
             capacity,
-            data,
+            data_hash,
             lock,
             type_,
+            ..
         } = self;
-        (capacity, data, lock, type_)
+        (capacity, data_hash, lock, type_)
     }
 
-    pub fn occupied_capacity(&self) -> CapacityResult<Capacity> {
-        Capacity::bytes(8 + self.data.len())
+    pub fn occupied_capacity(&self, data_bytes: u32) -> CapacityResult<Capacity> {
+        Capacity::bytes(8 + data_bytes as usize)
```

Comment:
I am not sure whether the occupied_capacity is right, now the data has been moved to the transaction.

Reply:
Yes, but rules of a cell is not changed.